test/python/algorithms/optimizers/test_spsa.py

```diff
@@ -201,9 +201,10 @@ def objective(x):
     def test_qnspsa_fidelity_deprecation(self):
         """Test using a backend and expectation converter in get_fidelity warns."""
         ansatz = PauliTwoDesign(2, reps=1, seed=2)
-
         with self.assertWarns(PendingDeprecationWarning):
-            _ = QNSPSA.get_fidelity(ansatz, StatevectorSimulatorPy(), MatrixExpectation())
+            QNSPSA.get_fidelity(ansatz, backend=StatevectorSimulatorPy())
```

Comment:
Positional args don't trigger the warning anymore. I will prioritize fixing that this week via https://github.com/Qiskit/qiskit-terra/issues/9865.

But I don't think we need to block landing this PR. It'd be good to land this so that we can unblock https://github.com/Qiskit/qiskit-terra/pull/9532

Reply:
In the case of Grover it allowed a Sampler to be passed positionally today where quantum_instance is (that would be the position it would be in the future once QI is removed - the typehinting did not show that though). I do not think other algorithms did that sort of thing preemptively. I do not know if that was taken advantage of anywhere.

Reply:
Grover's signature is this:

```python
    def __init__(
         self,
         iterations: Optional[Union[List[int], Iterator[int], int]] = None,
        growth_rate: Optional[float] = None,
        sample_from_iterations: bool = False,
        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
        sampler: Optional[BaseSampler] = None,
    ) -> None:
```

That means that to use positional arguments, you'd have to do `Grover(None, None, False, None, Sampler())`. Something like `Grover(Sampler())` would set the first argument `iterations` to be a `Sampler`, which is a bug fwict.

Reply:
What I meant was `Grover(None, None, False, Sampler()) ` will work here. Today this is the place where quantum_instance is, after quantum_instance is removed its the same place positionally. If you look at Grover it checks the type of quantum_instance at one point. As I mentioned I think this is the only one that supported this final positioning preemptively like this - it does not publish the fact in the typehints that this is possible.